File: 백준/Bronze/2460. 지능형 기차 2/지능형 기차 2.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int max = 0;
        int now = 0;
        for(int i = 0; i < 10; i++) {
            st = new StringTokenizer(br.readLine());
            int in = Integer.parseInt(st.nextToken());
            int out = Integer.parseInt(st.nextToken());
            
            now = now - in + out;
            
            if(now > max) max = now;
        }
        
        System.out.println(max);
    }
}

File: 백준/Bronze/2460. 지능형 기차 2/지능형 기차 2.kt
import java.io.*
import java.util.*

fun main(args : Array<String>) {
    var max = 0
    var now = 0

    for(i in 0 .. 9) {
        val list = readln().split(" ").map{it.toInt()}
        val come = list[0]
        val out = list[1]

        now = now - come + out

        if(now > max) max = now
    }

    println(max)
}